ui/src/app/pages/login/account-creation/account-creation-institution.tsx

```diff
@@ -0,0 +1,330 @@
+    return validate(this.state.profile, validationCheck);
+  }
+
+  getRoleOptions(): Array<{label: string, value: InstitutionalRole}> {
+    const {institutions, profile: {verifiedInstitutionalAffiliation: {institutionShortName}}} = this.state;
+    if (isBlank(institutionShortName)) {
+      return [];
+    }
+
+    const selectedOrgType = institutions.find(
+      inst => inst.shortName === institutionShortName).organizationTypeEnum;
+    const availableRoles: Array<InstitutionalRole> =
+      AccountCreationOptions.institutionalRolesByOrganizationType
+      .find(obj => obj.type === selectedOrgType)
+        .roles;
+
+    return AccountCreationOptions.institutionalRoleOptions.filter(option =>
+      availableRoles.includes(option.value)
+    );
+  }
+
+  updateAffiliationValue(attribute: string, value) {
+    this.setState(fp.set(['profile', 'verifiedInstitutionalAffiliation', attribute], value));
+  }
+
+  render() {
+    const {
+      loadingInstitutions,
+      institutions,
+      profile: {
+        contactEmail,
+        verifiedInstitutionalAffiliation: {
+          institutionShortName, institutionalRoleEnum, institutionalRoleOtherText,
+        }
+      }
+    } = this.state;
+
+    const errors = this.validate();
+
+    return <div id='account-creation-institution'
+                style={{paddingTop: '1.5rem', paddingRight: '3rem', paddingLeft: '3rem'}}>
+      <div style={{fontSize: 28, fontWeight: 400, color: colors.primary}}>Create your account</div>
+      <FlexRow>
+        <FlexColumn style={{marginTop: '0.5rem', marginRight: '2rem'}}>
+          <div style={{...styles.text, fontSize: 16, marginTop: '1rem'}}>
+            Please complete Step 1 of 3
+          </div>
+          <div style={{...styles.text, fontSize: 14, marginTop: '0.7rem'}}>
+            For access to the <i>All of Us</i> Research Program data, your institution needs to have signed a Data Use Agreement
+            with the program. The institutions listed below have an Institutional Data Use Agreement with the program that
+            enables us to provide their researchers with access to the Workbench.
+          </div>
+          <div style={{...styles.text, fontSize: 12, marginTop: '0.5rem'}}>
+            All fields are required.
+          </div>
+          {loadingInstitutions && <SpinnerOverlay />}
+          {!loadingInstitutions && <div style={{marginTop: '.5rem'}}>
+            <label style={{...styles.text, fontWeight: 600}}>
+              Select your institution
+              <i style={{...styles.publiclyDisplayedText, marginLeft: '0.2rem'}}>
+                Publicly displayed
+              </i>
+            </label>
+            <div style={{...styles.text, fontSize: 14}}>
+              Your institution will be notified that you have registered using your institutional credentials.
+            </div>
+            <Dropdown
+                data-test-id='institution-dropdown'
+                style={{width: '50%', minWidth: '600px'}}
```

Comment:
Can this be a style object? It is used multiple times

Reply:
Done